gatsby_fp/src/components/Layout/Body/Fields/IntellCont/SubHeading.js

```diff
@@ -58,21 +58,30 @@ export default function IntellCont({ intellContList, publicationType }) {
                       ? `${element.contypeother} -`
                       : ``
                   }
-                  ${
-                    element.intellcont_auth.lname
-                      ? `${element.intellcont_auth.lname},`
-                      : ``
-                  }
-                  ${
-                    element.intellcont_auth.mname
-                      ? `${element.intellcont_auth.lname},`
-                      : ``
-                  }
-                  ${
-                    element.intellcont_auth.fname
-                      ? `${element.intellcont_auth.fname}.`
-                      : ``
-                  }
+                  ${element.intellcont_auth.map(elem => (
+                    `
+                      ${
+                        elem.lname
+                          ? `${elem.lname},`
+                          : ``
+                      }
+                      ${
+                        elem.fname && !elem.mname
+                          ? `${elem.fname}.;`
+                          : ``
+                      }
+                      ${
+                        elem.mname && !elem.fname
+                          ? `${elem.mname}.;`
+                          : ``
+                      }
+                      ${
+                        elem.fname && elem.mname
+                          ? `${elem.fname}. ${elem.mname}.;`
+                          : ``
+                      }
+                    `
+                  )).join('')}
```

Comment:
What about replacing it with this 
`${element.intellcont_auth.map(elem => (
                  `
                  ${elem.lname ?? ''}, ${elem.fname ?? ''} ${elem.mname ?? ''};
                  `
                )).join('')}`

Reply:
The reason I have multiple cases is because I didn't know if mname was guaranteed to exist @dmotan

Reply:
I see, and it's never guaranteed! Are you familiar with [nullish coalescing operator (??)](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Nullish_coalescing_operator)?
When I say `elem.mname ?? ''` this means if elem.mname exisits, then use it's value, otherwise use ''

Reply:
No i actually was not familiar with that syntax, but in terms of consistency, if I changed this specific diff of code would we want the rest to be the same syntax? @dmotan

Reply:
Good point, I'd rather be consistent over using a new syntax.
I'd remove the dots after names, curious to see what Mike thinks though!

Reply:
Gotcha, yeah the main reason why I have the dots after the names is because Nikhil indicated in the formatting document that he wanted them typed in, some users are getting double periods after their mname because they typed in a period after the initial @dmotan @mf-stevens

Reply:
Isn't the spec to truncate `fname` and `mname` each to 1 character and then for whichever one(s) exist add a dot after it?

Reply:
Also, ternary is most useful when you actually have an else situation. Using a ternary with nothing but an empty string as the else value is code smell suggesting that you shouldn't be using a ternary at all; you can simply use a normal `if` statement which allows you to fully omit the `else` if you want. If you convert from an implicit return to an explicit return, you'll be able to do all the truncation and so forth without being confined to template literals.

Reply:
@davidwatson Shoot some holes in the above ;)

Reply:
Less hole-shooting, and more light reframing: The heuristic to shoot for here is "Is this code readable and maintainable _by the team as a whole_?"

To that end, I'm okay with a few minor inconsistencies in this project. This is every bit as much a growth opportunity as a deliverable, and we can smooth them over post-launch. There's something to be said for consistency, though now is a good time to push for ideal solutions so we can iterate on them now and replicate them later.

More to the point: Personally, I agree that using a ternary here probably isn't the most expressive. I think nullish coalescing is a great solution to the problem, but I'm also used to it. Similarly, I tend to use functional techniques more frequently and shy away from control statements—but again, my own individual style isn't what matters here. We should be taking this time to discuss which will be the most natural and expressive to read and write in the future, for all of us.